test/x509_time_test.c

```diff
@@ -424,11 +424,71 @@ static int test_days(int n)
     return r;
 }
 
+#define construct_asn1_time(s, t, e) \
+    { { sizeof(s) - 1, t, (unsigned char*)s, 0 }, e }
+
+static const struct {
+    ASN1_TIME asn1;
+    const char *readable;
+} x509_print_tests [] = {
+    /* Generalized Time */
+    construct_asn1_time("20170731222050Z", V_ASN1_GENERALIZEDTIME,
+            "Jul 31 22:20:50 2017 GMT"),
+    /* Generalized Time, no seconds */
+    construct_asn1_time("201707312220Z", V_ASN1_GENERALIZEDTIME,
+            "Jul 31 22:20:00 2017 GMT"),
+    /* Generalized Time, fractional seconds (3 digits) */
+    construct_asn1_time("20170731222050.123Z", V_ASN1_GENERALIZEDTIME,
+            "Jul 31 22:20:50.123 2017 GMT"),
+    /* Generalized Time, fractional seconds (1 digit) */
+    construct_asn1_time("20170731222050.1Z", V_ASN1_GENERALIZEDTIME,
+            "Jul 31 22:20:50.1 2017 GMT"),
+    /* Generalized Time, fractional seconds (0 digit) */
+    construct_asn1_time("20170731222050.Z", V_ASN1_GENERALIZEDTIME,
+            "Bad time value"),
+    /* UTC Time */
+    construct_asn1_time("170731222050Z", V_ASN1_UTCTIME,
+            "Jul 31 22:20:50 2017 GMT"),
+    /* UTC Time, no seconds */
+    construct_asn1_time("1707312220Z", V_ASN1_UTCTIME,
+            "Jul 31 22:20:00 2017 GMT"),
+};
+
+static int test_x509_time_print(int idx)
+{
+    BIO *m;
+    int ret = 0, rv;
+    char *pp;
+    const char *readable;
+
+    if (!TEST_ptr(m = BIO_new(BIO_s_mem())))
+        goto err;
+    if (!TEST_int_ge(rv = ASN1_TIME_print(m, &x509_print_tests[idx].asn1), 0)) {
```

Comment:
This is mute test, because ASN1_TIME_print doesn't return negative values, i.e. returns either 1 or 0.